### PR TITLE
Httpclient updates

### PR DESCRIPTION
**Pull request by bamarni**
- add a default transport
- set a default timeout of 3 minutes
- fix a flaky test
- add a functional option type for Requests

### CODE REVIEW

**Comment by bamarni** [comment]
run windows integration tests

File: pkg/httpclient/client_test.go [hunk 1]
```diff
@@ -59,39 +60,70 @@ func TestPost(t *testing.T) {
...
+func TestNewRequestWithoutTimeout(t *testing.T) {
+	client := New("https://dcos.io", func(client *Client) {
+		client.timeout = 60 * time.Second
+	})
+
+	req, err := client.NewRequest("GET", "/path", nil, NoTimeout())
+	require.NoError(t, err)
+	deadline, ok := req.Context().Deadline()
+	require.False(t, ok)
+	require.True(t, deadline.IsZero())
```
Comment (jdef) [inline pkg/httpclient/client_test.go]: why bother testing the value of `deadline` if `ok` is false?

File: pkg/httpclient/client_test.go [hunk 2]
```diff
@@ -59,39 +60,70 @@ func TestPost(t *testing.T) {
...
+}
+
+func TestCancelRequest(t *testing.T) {
+	done := make(chan struct{})
+	stuckedHandler := make(chan struct{})
```
Comment (jdef) [inline pkg/httpclient/client_test.go]: nit: s/stuckedHandler/stuckHandler/

File: pkg/httpclient/client.go [hunk 3]
```diff
@@ -46,10 +85,18 @@ func (c *Client) Post(path string, contentType string, body io.Reader) (*http.Re
 	return c.Do(req)
 }
 
+// NoTimeout disables the timeout for an HTTP request.
+func NoTimeout() RequestOption {
+	return func(req *http.Request) {
+		ctx := context.WithValue(req.Context(), keyNoTimeout, true)
```
Comment (jdef) [inline pkg/httpclient/client.go]: nit: I probably would have used `struct{}{}` instead of `true` since a value is not actually needed here

**Pull request merged**